airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/test_source.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from source_amazon_seller_partner import SourceAmazonSellerPartner
+from source_amazon_seller_partner import ConnectorConfig
+from airbyte_cdk.models import ConnectorSpecification
+from airbyte_cdk.sources.streams import Stream
+
+
+@pytest.fixture
+def connector_source():
+    return SourceAmazonSellerPartner()
+
+
+@pytest.fixture
+def connector_config():
+    return ConnectorConfig(
+        replication_start_date="2017-01-25T00:00:00Z",
+        refresh_token="Atzr|IwEBIP-abc123",
+        lwa_app_id="amzn1.application-oa2-client.abc123",
+        lwa_client_secret="abc123",
+        aws_access_key="aws_access_key",
+        aws_secret_key="aws_secret_key",
+        role_arn="arn:aws:iam::123456789098:role/some-role",
+        aws_environment="SANDBOX",
+        region="US"
+    )
+
+
+class FakeBotoClient:
+    role = {
+        "Credentials": {
+            "AccessKeyId": "foo",
+            "SecretAccessKey": "bar",
+            "SessionToken": "foobar",
+        }
+    }
+
+    def assume_role(self, *args, **kwargs):
+        return self.role
+
+    def get_session_token(self, *args, **kwargs):
+        return self.role
+
+
+@pytest.fixture
+def mock_boto_client(mocker):
+    mocker.patch("boto3.client", return_value=FakeBotoClient())
+
+
+def test_spec(connector_source):
+    assert isinstance(connector_source.spec(), ConnectorSpecification)
+
+
+def test_streams(connector_source, connector_config, mock_boto_client):
+    for stream in connector_source.streams(connector_config):
+        assert isinstance(stream, Stream)
+
+
+@pytest.mark.parametrize(
+    "arn",
+    ("arn:aws:iam::123456789098:user/some-user", "arn:aws:iam::123456789098:role/some-role")
+)
+def test_stream_with_good_iam_arn_value(connector_source, connector_config, mock_boto_client, arn):
+    connector_config.role_arn = arn
+    result = connector_source.get_sts_credentials(connector_config)
+    assert "Credentials" in result
```

Comment:
I still find this test a bit weak as it's not checking that something different happens for role and user. Please assert specific function calls on a mocked client

Reply:
sorry I don't understand what you mean 😔, can you help me with this please?

Reply:
```

from source_amazon_seller_partner import source

@pytest.mark.parametrize(
    "arn",
    ("arn:aws:iam::123456789098:user/some-user", "arn:aws:iam::123456789098:role/some-role")
)
def test_stream_with_good_iam_arn_value(mocker, connector_source, connector_config, arn):
    mocker.path.object(source, "boto3.client")
    mock_boto_client = source.boto3.client.return_value
    connector_config.role_arn = arn
    result = connector_source.get_sts_credentials(connector_config)
    assert "Credentials" in result
    if "user" in arn:
        mock_boto_client.get_session_token.assert_called_once()
    if "role" in arn:
        mock_boto_client.assume_role.assert_called_once_with(RoleArn=arn, RoleSessionName="guide")
```

Reply:
Thanks 🙏 . I learned things with this.